packages/ckeditor5-editor-multi-root/src/augmentation.ts

```diff
@@ -79,5 +79,16 @@ declare module '@ckeditor/ckeditor5-core' {
 		 * ```
 		 */
 		rootsAttributes?: Record<string, RootAttributes>;
+
+		/**
+		 * List of names of all the roots that exist in the document but are not initially loaded by the editor.
+		 *
+		 * These roots can be loaded at any time after the editor has been initialized, using
+		 * {@link module:editor-multi-root/multirooteditor~MultiRootEditor#loadRoot `MultiRootEditor#lazyRoot()`}.
+		 *
+		 * This is useful for handling big documents that contain hundreds of roots, or contain very large roots, which may have
+		 * impact editor performance if loaded all together.
+		 */
+		lazyRoots?: Array<string>;
```

Comment:
I remember our discussion, sorry for not giving you a heads up. Basically, I plan even bigger refactoring with even more options coming in next releases. This is why I didn't want to introduce breaking change right now, as there might be another breaking change in the next release. So, I opted to wait for upcoming improvements to decide on final config API, instead of potentatially doing multiple breaking changes.